### PR TITLE
[CI-8134] print command before step execution

### PR DESCRIPTION
**Pull request by devkimittal**
…mmands

import ordering

removes logic to print out set -x

### CODE REVIEW

File: engine/exec/exec.go [hunk 1]
```diff
@@ -27,6 +32,16 @@ func Run(ctx context.Context, step *spec.Step, output io.Writer) (*runtime.State
 	cmd.Env = toEnv(step.Envs)
 	cmd.Stderr = output
 	cmd.Stdout = output
+	if !isDrone && len(step.Command) > 0 {
+		stepCommand := strings.TrimSpace(strings.Join(step.Command, ""))
+		if stepCommand != "" {
+			printCommand := "Executing the following command(s):\n" + stepCommand
+			lines := strings.Split(printCommand, "\n")
```
Comment (eoinmcafee00) [inline engine/exec/exec.go:39]: Splitting on '/n' isn't going to work for all cases.

**Comment by eoinmcafee00** [comment]
@devkimittal I'd like to see unit test cases .

File: engine/exec/exec.go (hunk 1 above)
Comment (devkimittal) [reply engine/exec/exec.go:39]: that's what i thought, but it was working, i tried with putting \n, \\n in the command

**Comment by devkimittal** [comment]
@eoinmcafee00 worked alright, the command
Screenshot 2023-08-09 at 4 10 41 PM
Screenshot 2023-08-09 at 4 10 07 PM
 comes as escaped string.

**Comment by eoinmcafee00** [comment]
> …mmands
> 
> import ordering
> 
> removes logic to print out set -x
Fix the commit message please.

File: engine/docker/docker.go [hunk 2]
```diff
@@ -183,12 +185,23 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
...
 func (e *Docker) Run(ctx context.Context, pipelineConfig *spec.PipelineConfig, step *spec.Step,
-	output io.Writer) (*runtime.State, error) {
+	output io.Writer, isDrone bool) (*runtime.State, error) {
 	// create the container
 	err := e.create(ctx, pipelineConfig, step, output)
 	if err != nil {
 		return nil, errors.TrimExtraInfo(err)
 	}
+
+	if !isDrone && len(step.Command) > 0 {
```
Comment (vistaarjuneja) [inline engine/docker/docker.go:195]: could we move this to the caller of Run? Trying to move stuff out of engine/docker as it's pretty simple at the moment

Comment (devkimittal) [reply engine/docker/docker.go:195]: good suggestion will do

**Comment by devkimittal** [comment]
Jira : https://harness.atlassian.net/browse/CI-8134

File: engine/engine.go [hunk 3]
```diff
@@ -122,13 +123,27 @@ func (e *Engine) Run(ctx context.Context, step *spec.Step, output io.Writer) (*r
 		vol.Path = pathConverter(vol.Path)
 	}
 
+	printCommand(step, output, isDrone)
```
Comment (tphoney) [inline engine/engine.go:126]: i would refactor this a little, as it doesnt follow the look and feel in the rest of this function

ie 
```
if !isDrone && len(step.Command) > 0 {
   printCommand(step, output)
end
```
then remove the conditional from the function

**Review by tphoney — COMMENTED** [review]
just to make it more readable i added a subtle change

**Review by vistaarjuneja — APPROVED** [review]

**Pull request merged**